### PR TITLE
Fix ENV parsing

### LINKED ISSUE #151: Environment variable FASTIFY_WATCH is not honored
The reason why is because `require('dotenv').config()` is not called inside the `start()` function, and therefore

```js
if (opts.watch) {
  return watch(args, opts.ignoreWatch)
}
```

is never true, unless there is an actual CLI argument set for it.

Unfortunately putting `require('dotenv').config()` on top of the `start()` function breaks your test suite. So probably there is something else going on that I'm not aware of.

---

My `.env` file looks like this:

```bash
FASTIFY_WATCH=true
```

And I'm starting the server with:

```bash
fastify start server.js
```

### PR DESCRIPTION
**Pull request by cemremengu**
Fixes #151 

Not sure how I if I can add a test case for this 

cc @mcollina is there an effective way to test this?

### CODE REVIEW

**Comment by mcollina** [comment]
I don't understand the problem. What is the problem?

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "lint": "standard",
     "unit": "tap test/*.test.js templates/app/test/*/*.test.js",
-    "test": "npm run lint && npm run unit"
+    "test": "npm run lint && npm run unit --require dotenv/config"
```
[the remark this answers is not in the document]
Comment (cemremengu) [reply package.json]: No, that was necessary in fact

**Comment by mcollina** [comment]
You can likely do a test case by using `child_process` to kick off a new process with the related config.

**Comment by cemremengu** [comment]
> no need to require the "dev" module in code. In production, you will use ENV only.

@mcollina maybe we should remove it altogether as suggested by @StarpTech?

**Comment by mcollina** [comment]
Supporting dotenv is transparent in production but it greatly simplifies the development experience. It's a development experience tool, not a production tool.

I still don't understand what is the issue here.

**Comment by simov** [comment]
Hi @mcollina, I raised the issue here https://github.com/fastify/fastify-cli/issues/151 The problem is that the documentation states that `FASTIFY_WATCH` can be used in place of `--watch` but in fact that's not possible. Is there something that I'm missing?

**Comment by simov** [comment]
Probably the documentation should be updated in case `--watch` is not supposed to be used as env var?

**Comment by cemremengu** [comment]
Yes maybe we should just update the docs. It doesnt make sense to have watch as env variable I think.

**Comment by cemremengu** [comment]
@mcollina the problem is `watch` env variable is not parsed due to the location of `require('dotenv').config()`

**Comment by mariusa** [comment]
This happens again. This should be moved back to top?
https://github.com/fastify/fastify-cli/blob/master/start.js#L62

**Comment by mcollina** [comment]
I'm not sure why it was moved down again. Could you open a fresh issue?

**Pull request merged**